analyzers/VMRay/vmray.py

```diff
@@ -7,98 +7,230 @@
-        predicate = "Scan"
-
-        r = {
-            'reports': []
-        }
-
-        if raw.get('scanreport', None) and len(raw.get('scanreport').get('data')) > 0:
-            for scan in raw.get('scanreport').get('data'):
-                r['reports'].append({
-                    'score': scan.get('sample_score'),
-                    'sample_severity': scan.get('sample_severity'),
-                    'sample_last_reputation_severity': scan.get('sample_last_reputation_severity'),
-                    'url': scan.get('sample_webif_url')
-                })
-
-        if len(r["reports"]) == 0:
-            value = "No Scan"
-            level = "info"
-            taxonomies.append(self.build_taxonomy(level, namespace, predicate, value))
+        samples = raw.get("samples", [])
+        if len(samples) == 0:
+            taxonomies.append(
+                self.build_taxonomy("info", self._namespace, "None", "No Scan")
+            )
         else:
-            for s in r["reports"]:
-                i = 1
-                if s["sample_severity"] == "not_suspicious":
-                    level = "safe"
-                elif s["sample_severity"] == "malicious":
-                    level = "malicious"
-                else:
-                    level = "info"
-
-                if len(r["reports"]) > 1:
-                    value = "{}( from scan {})".format(s["score"], i)
-                else:
-                    value = "{}".format(s["score"])
-                taxonomies.append(self.build_taxonomy(level, namespace, predicate, value))
-                i += 1
-
+            taxonomies.extend(self._taxonomies_for_samples(samples))
         return {"taxonomies": taxonomies}
 
+    def _artifacts_for_samples(self, samples):
+        artifacts = []
+        for sample in samples:
+            link = sample.get("sample_webif_url", None)
+            iocs = sample.get("sample_iocs", {}).get("iocs", {})
+
+            for ioc_type, ioc_payload_name in self._ioc_mapping.items():
+                if ioc_type in iocs:
+                    for ioc_node in iocs[ioc_type]:
+                        severity = ioc_node.get("severity", "")
+                        level = self._severity_mapping.get(severity, "info")
+                        tags = [severity, level, ioc_node["type"]]
+                        payload = ioc_node[ioc_payload_name]
+                        if "hashes" in ioc_node:
+                            for hash_node in ioc_node["hashes"]:
+                                if "sha256_hash" in hash_node:
+                                    payload += ", SHA256: {}".format(
+                                        hash_node["sha256_hash"]
+                                    )
+                        elif "operations" in ioc_node:
+                            payload += ", Operations: " + ", ".join(
+                                ioc_node["operations"]
+                            )
+                        artifacts.append(
```

Comment:
A single hash or a single URL maps to one artifact, eg. an artifact itself does not contain a list as payload. Artifacts is a list of hashes, URLs, etc. with some meta data attached to them.